src/test/java/cucumber/stepdefs/GreetingServiceStepDefs.java

```diff
@@ -27,6 +28,15 @@ public GreetingServiceStepDefs(){
             }
         });
 
+        When("^the client requests /GET /greeting using query string parameter User$", () -> {
+            try {
+                greetingServiceSteps.theClientRequestsGETGreetingUsingQueryStringParameterUser();
+            } catch (Throwable t) {
+                System.out.println("Throwable caught");
```

Comment:
Use capture group to pass parameter to method. Note: use (.*) to capture a String.